### ISSUE TITLE
[BUG] swanlab: Data type error, key: eval_normalized_wer, data type: NoneType, expected: float

### ISSUE THREAD

**Issue by 80boys** [SwanHubX/SwanLab#760]
```python
swanlab.init(
    workspace="wuheng0319",
    project="finetuning",
    config={
    "learning_rate": 1e-5,
    "batch_size": 8,
    "epochs": 2,
    "gradient_accumulation_steps": 1,
    }
)

wandb.login(key=WANDB_TOKEN)
wandb.init(project="finetuning", name="training-run", config={
    "learning_rate": 1e-5,
    "batch_size": 8,
    "epochs": 2,
    "gradient_accumulation_steps": 1,
})

```

![image](https://github.com/user-attachments/assets/a4d311f9-aaa1-4e48-9e7d-45d2296c81b0)

![image](https://github.com/user-attachments/assets/da23713f-5966-40e6-b723-c8a6edb2e8e5)

```

swanlab: Tracking run with swanlab version 0.4.2
swanlab: Run data will be saved locally in /home/utalk/jupyter/TuneWhisper/swanlog/run-20241227_185531-004ddc0c
swanlab: 👋 Hi wuheng0319, welcome to swanlab!
swanlab: Syncing run ox-3 to the cloud
swanlab: 🌟 Run `swanlab watch /home/utalk/jupyter/TuneWhisper/swanlog` to view SwanLab Experiment Dashboard locally
swanlab: 🏠 View project at https://swanlab.cn/@wuheng0319/whisper-finetuning
swanlab: 🚀 View run at https://swanlab.cn/@wuheng0319/whisper-finetuning/runs/sf125vde6npyd78utgc6v
trainable params: 15,728,640 || all params: 1,559,033,600 || trainable%: 1.0088711365810203
swanlab: Data type error, key: eval_loss, data type: NoneType, expected: float
swanlab: Chart 'eval_loss' creation failed. Reason: The expected value type for the chart 'eval_loss' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Data type error, key: eval_wer, data type: NoneType, expected: float
swanlab: Chart 'eval_wer' creation failed. Reason: The expected value type for the chart 'eval_wer' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Data type error, key: eval_normalized_wer, data type: NoneType, expected: float
swanlab: Chart 'eval_normalized_wer' creation failed. Reason: The expected value type for the chart 'eval_normalized_wer' is one of int,float or BaseType, but the input type is NoneType.
{'loss': 5.9328, 'grad_norm': 4.758566379547119, 'learning_rate': 2.0000000000000002e-07, 'epoch': 0.0}
swanlab: Chart 'eval_loss' creation failed. Reason: The expected value type for the chart 'eval_loss' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_wer' creation failed. Reason: The expected value type for the chart 'eval_wer' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_normalized_wer' creation failed. Reason: The expected value type for the chart 'eval_normalized_wer' is one of int,float or BaseType, but the input type is NoneType.
{'loss': 5.1045, 'grad_norm': 4.454663276672363, 'learning_rate': 1.8000000000000001e-06, 'epoch': 0.01}
swanlab: Chart 'eval_loss' creation failed. Reason: The expected value type for the chart 'eval_loss' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_wer' creation failed. Reason: The expected value type for the chart 'eval_wer' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_normalized_wer' creation failed. Reason: The expected value type for the chart 'eval_normalized_wer' is one of int,float or BaseType, but the input type is NoneType.
{'loss': 5.4229, 'grad_norm': 4.392512798309326, 'learning_rate': 3.4000000000000005e-06, 'epoch': 0.02}
swanlab: Chart 'eval_loss' creation failed. Reason: The expected value type for the chart 'eval_loss' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_wer' creation failed. Reason: The expected value type for the chart 'eval_wer' is one of int,float or BaseType, but the input type is NoneType.
swanlab: Chart 'eval_normalized_wer' creation failed. Reason: The expected value type for the chart 'eval_normalized_wer' is one of int,float or BaseType, but the input type is NoneType.
{'loss': 5.1348, 'grad_norm': 4.625607013702393, 'learning_rate': 5.400000000000001e-06, 'epoch': 0.02}

```
我不知道问题出在哪里，但是 wandb 是好用的，而且人家还支持 代码登录 有login 方法

**Comment by 80boys** [comment]
`class CustomTrainerCallback(TrainerCallback):
    def on_log(self, args, state, control, **kwargs):
        logs = kwargs.get("logs")
        if logs is not None:
            step = state.global_step
            loss = logs.get("loss")
            eval_loss = logs.get("eval_loss")
            eval_wer = logs.get("eval/wer")
            eval_normalized_wer = logs.get("eval/normalized_wer")

            swanlab.log({
                "train_loss": loss,
                "eval_loss": eval_loss,
                "eval_wer": eval_wer,
                "eval_normalized_wer": eval_normalized_wer,
                "step": step
            })
            
            wandb.log({
                "train_loss": loss,
                "eval_loss": eval_loss,
                "eval_wer": eval_wer,
                "eval_normalized_wer": eval_normalized_wer,
                "step": step
            })
        return control
        
    `
    
    上报代码如上

**Comment by Zeyi-Lin** [comment]
Hi，我们在跟踪这个问题
此外，swanlab也支持login方法，方式为：

```python
swanlab.login(api_key="[你的API Key]")
```

**Comment by SAKURA-CAT** [comment]
> {'loss': 5.9328, 'grad_norm': 4.758566379547119, 'learning_rate': 2.0000000000000002e-07, 'epoch': 0.0}

主要您的`eval_normalized_wer `并没有在此处打印，我并不清楚他的具体类型是什么。wandb可能考虑的多一些，您愿意提供更多的类型信息吗？

**Comment by SAKURA-CAT** [comment]
此外根据我的观察，您对比的并不是同一份数据，wandb自动的将`eval_normalized_wer`归组到了`eval/normalized_wer `（swanlab没有这个功能，因为我不确定这是否有必要），因此您上传的wandb的图像似乎全部和train相关的。

![image](https://github.com/user-attachments/assets/cf67c309-9218-482e-9b6d-eedfec2c8f14)
![image](https://github.com/user-attachments/assets/ca02e56d-b62f-4ccc-8e93-19f2de857c7f)
